src/Facade/Services/SupplierFacadeService.cs

```diff
@@ -126,8 +127,29 @@ private static Supplier BuildEntityFromResourceHelper(SupplierResource resource)
                                         ? DateTime.Parse(resource.DateClosed) : null,
                            ReasonClosed = resource.ReasonClosed,
                            Notes = resource.Notes,
-                           OrganisationId = resource.OrganisationId
-            };
+                           OrganisationId = resource.OrganisationId,
+                           Contacts = resource.Contacts?.Select(c => new SupplierContact
+                                                                          {
+                                                                              SupplierId = resource.Id,
+                                                                              ContactId = c.Id > 0 ? c.Id : this.databaseService.GetIdSequence("CONT_SEQ"),
```

Comment:
i think the id should be set in the persistence layer but I can see that it in theses instances it would require the creation of whole new repository classes just for that single reason so this does seem a reasonable balance